### PR TITLE
🐛 [amp-list] Call scheduleUnlayout when amp-list container is updated

### PR DESCRIPTION
**Pull request by krdwan**
Fixes: https://github.com/ampproject/amphtml/issues/21680

Issue: 
- When a component is used within amp-list, and amp-list re-renders its list contents, the previous elements do not have their `unlayoutCallback()` (amp lifecycle callback) called.
- In the issue linked above, `amp-bind` with a new `src` triggers the re-render
- `Amp-date-countdown` uses `setInterval()` to periodically update its date.  It then does `clearInterval` in it's `unlayoutCallback()` when it is no longer used
- Since the `unlayoutCallback()` is not called when `amp-date-countdown` is used within `amp-list`, after `amp-list` re-renders, the intervals is still set and continues to run it's callback
- In this situation, the callback tries to access the root node of the element (which is no longer the DOM's root) and  repeatedly errors out when it cannot do `getElementById()`

Fix:
- Call `scheduleUnlayout` on removed elements from `amp-list` (which runs `unlayoutCallback()`) each time it re-renders
- In addition to `amp-date-countdown` this should also fix any undiscovered bugs related to amp-elements not being cleaned up with `unlayoutCallback()` when used within `amp-list`

### CODE REVIEW

File: extensions/amp-list/0.1/amp-list.js [hunk 1]
```diff
@@ -1055,6 +1059,9 @@ export class AmpList extends AMP.BaseElement {
         this.diff_(container, elements);
       } else {
```
[the remark this answers is not in the document]
Comment (krdwan) [reply extensions/amp-list/0.1/amp-list.js:1067]: Thanks, yep Caroline, I've added in this latest version!

File: extensions/amp-list/0.1/amp-list.js [hunk 2]
```diff
@@ -602,6 +606,14 @@ export class AmpList extends AMP.BaseElement {
...
           });
         }
+        toArray(
+          dev().assertElement(this.container_).children
+        ).forEach((child) =>
+          this.owners_./*OK*/ scheduleUnlayout(
+            dev().assertElement(this.container_),
+            child
+          )
+        );
```
Comment (caroqliu) [inline extensions/amp-list/0.1/amp-list.js:616]: Under the hood `removeChildren` iterates through all the children already, so let's replace the call to `removeChildren` with similar logic:
```js
while (parent.firstChild) {
  this.owners_./*OK*/ scheduleUnlayout(
    dev().assertElement(parent),
    child
  )
  parent.removeChild(parent.firstChild);
}
```

File: extensions/amp-list/0.1/amp-list.js [hunk 3]
```diff
@@ -602,6 +606,14 @@ export class AmpList extends AMP.BaseElement {
             'update': false,
           });
         }
+        toArray(
+          dev().assertElement(this.container_).children
+        ).forEach((child) =>
+          this.owners_./*OK*/ scheduleUnlayout(
+            dev().assertElement(this.container_),
```
Comment (jridgewell) [inline extensions/amp-list/0.1/amp-list.js:613]: This must be the owning amp-element, not a regular `<div>`. So, the `<amp-list>`?

File: extensions/amp-list/0.1/amp-list.js [hunk 4]
```diff
@@ -602,6 +606,14 @@ export class AmpList extends AMP.BaseElement {
...
+            child
```
Comment (jridgewell) [inline extensions/amp-list/0.1/amp-list.js:614]: You'd want to pass the `container_` here, and we'll find all amp-element under that container.

**Pull request merged**